File: heavy-metal-bake-sale/typescript/tests/OrderBuilder.ts
import { Money } from '../src/Money.js';
import { Product } from '../src/Product.js';
import { BakeSale } from '../src/BakeSale.js';

export class OrderBuilder {
  private readonly _inventory: Product[] = [];

  withProduct(product: Product): this {
    this._inventory.push(product);
    return this;
  }

  withDefaultInventory(): this {
    this._inventory.push(new Product('Brownie', new Money(0.75), 'B', 48));
    this._inventory.push(new Product('Muffin', new Money(1.00), 'M', 36));
    this._inventory.push(new Product('Cake Pop', new Money(1.35), 'C', 24));
    this._inventory.push(new Product('Water', new Money(1.50), 'W', 30));
    return this;
  }

  build(): BakeSale {
    return new BakeSale(this._inventory);
  }
}

File: heavy-metal-bake-sale/python/tests/order_builder.py
from __future__ import annotations

from heavy_metal_bake_sale import BakeSale, Money, Product


class OrderBuilder:
    def __init__(self) -> None:
        self._inventory: list[Product] = []

    def with_product(self, product: Product) -> OrderBuilder:
        self._inventory.append(product)
        return self

    def with_default_inventory(self) -> OrderBuilder:
        self._inventory.append(Product("Brownie", Money(0.75), "B", 48))
        self._inventory.append(Product("Muffin", Money(1.00), "M", 36))
        self._inventory.append(Product("Cake Pop", Money(1.35), "C", 24))
        self._inventory.append(Product("Water", Money(1.50), "W", 30))
        return self

    def build(self) -> BakeSale:
        return BakeSale(self._inventory)
